core/extension/gdextension_manager.cpp

```diff
@@ -236,6 +236,18 @@ void GDExtensionManager::load_extensions() {
 	}
 
 	OS::get_singleton()->load_platform_gdextensions();
+
+	if (p_init_func) {
+		Ref<GDExtension> libgodot;
+		libgodot.instantiate();
+		Error err = libgodot->initialize_extension_function(p_init_func, "lib_godot");
+		if (err != OK) {
+			ERR_PRINT("Error initializing extension function");
+		} else {
+			libgodot->set_path("res://__LibGodot");
+			GDExtensionManager::get_singleton()->load_extension("res://__LibGodot");
```

Comment:
This PR introduces a bunch of new names, and, while I don't necessarily want to start bikeshedding, we probably should discuss if these are the names we want to use, and, in particular, are they internally consistent, and consistent with names already used in other parts of Godot.

So, here we have:

- `"lib_godot"` as the fake entry point name
- `"res://__LibGodot"` as the fake library path. (Also, this needs to be unique among all extensions - with this PR, are developers able to register only one faux GDExtension from their application or multiple? In @Faolan-Rad's PR, if I'm remembering correctly, more than one faux GDExtension could be registered. If multiple are allowed, we'll need a unique path for each.)

And elsewhere we have:

- The `LIBRARY_ENABLED` define
- The "Embedded" feature tag
- The general term "embedded" to describe a faux GDExtension registered through libgodot

These seem a little bit inconsistent, using the terms "libgodot", "embedded" and "library" in different places.

Reply:
I agree, and we are open to naming suggestions. I agree that it is important to have consistent naming. Should we just try to use LIBGODOT everywhere as the key for this? Or should we move to the more generic "LIBRARY" name?

Reply:
Coming back to this because it seems there's still some mixed names (although, it's _much_ improved from the last time I went through this PR).

Personally, I don't like the super generic "library".

I'd be for either "LibGodot" or "embedded Godot", since they are more descriptive, with maybe a slight preference for "LibGodot" because that sounds more like a product/feature name, making it easier to refer to.

If we did go for "LibGodot", then we'd end up with a `LIBGODOT_ENABLED` define, `libgodot://` path, `libgodot` feature tag, and "LibGodot" in the docs and other human-readable places.

I'm surprised there hasn't been more bike-shedding about the name :-)